server/game/costs/ReduceableFateCost.js

```diff
@@ -1,102 +1,40 @@
-    }
-}
-
 class ReduceableFateCost {
     constructor(playingType) {
         this.playingType = playingType;
     }
 
     canPay(context) {
-        let minCost = context.player.getMinimumCost(this.playingType, context.source);
+        let minCost = context.player.getMinimumCost(this.playingType, context);
         return context.player.fate >= minCost && (minCost === 0 || context.player.checkRestrictions('spendFate', context));
     }
 
     resolve(context, result) {
-        //console.log('resolveCost');
         let alternatePools = context.player.getAlternateFatePools(this.playingType, context.source);
         let alternatePoolTotal = alternatePools.reduce((total, pool) => total + pool.fate, 0);
         let maxPlayerFate = context.player.checkRestrictions('spendFate', context) ? context.player.fate : 0;
-        let minCostReduction = Math.max(this.getReducedCost(context) - maxPlayerFate - alternatePoolTotal, 0);
-        context.game.queueStep(new FateCostInterruptWindow(context, minCostReduction, result));
-        context.game.queueSimpleStep(() => {
-            alternatePoolTotal = alternatePools.reduce((total, pool) => total + pool.fate, 0);
-            if(this.getReducedCost(context) > maxPlayerFate + alternatePoolTotal) {
-                result.cancelled = true;
-            } else if(!result.cancelled && alternatePools.length > 0 && context.player.checkRestrictions('takeFateFromRings', context)) {
-                let properties = {
-                    reducedCost: this.getReducedCost(context),
-                    remainingPoolTotal: alternatePoolTotal
-                };
-                context.costs.alternateFate = new Map();
-                let maxPlayerFate = context.player.checkRestrictions('spendFate', context) ? context.player.fate : 0;
-                for(const alternatePool of alternatePools) {
-                    context.game.queueSimpleStep(() => {
-                        properties.remainingPoolTotal -= alternatePool.fate;
-                        properties.minFate = Math.max(properties.reducedCost - maxPlayerFate - properties.remainingPoolTotal, 0);
-                        properties.maxFate = Math.min(alternatePool.fate, properties.reducedCost);
-                        properties.pool = alternatePool;
-                        properties.numberOfChoices = properties.maxFate - properties.minFate + 1;
-                        if(result.cancelled || properties.numberOfChoices === 0) {
-                            return;
-                        }
-                        this.promptForAlternateFate(context, result, properties);
-                    });
-                }
+        if(this.getReducedCost(context) > maxPlayerFate + alternatePoolTotal) {
+            result.cancelled = true;
+        } else if(!result.cancelled && alternatePools.length > 0 && context.player.checkRestrictions('takeFateFromRings', context)) {
```

Comment:
Seems a bit strange to me that you would be checking for 'takeFateFromRings' restrictions here.  Correct me if I'm wrong but I would have thought the idea was to be able to have different alternate fate pools that aren't necessarily from the fate rings.

Reply:
Yeah, I kind of left it open so that non-ring fate pools fit easily in in the future.  However, for the moment, they don't exist, and it didn't seem worth doing extra work to distinguish between until that changes

Reply:
Sure.  Just something to watch out for in the future.  I could certainly see them printing a holding that builds up fate or something.  Actually, that sounds like a cool idea for a card :)